checker/mod_checking.ml

```diff
@@ -17,9 +17,12 @@ let set_indirect_accessor f = indirect_accessor := f
 
 let check_constant_declaration env kn cb =
   Flags.if_verbose Feedback.msg_notice (str "  checking cst:" ++ Constant.print kn);
-  (* Locally set the oracle for further typechecking *)
-  let oracle = env.env_typing_flags.conv_oracle in
-  let env = Environ.set_oracle env cb.const_typing_flags.conv_oracle in
+  (* Locally set typing flags for further typechecking *)
+  let orig_flags = env.env_typing_flags in
+  let cb_flags = cb.const_typing_flags in
+  let env = Environ.set_typing_flags {orig_flags with check_guarded = cb_flags.check_guarded;
```

Comment:
This should be in the `let env'`, such that you don't need to reset typing flags.

Reply:
This is still todo

Reply:
Indeed. Sorry for having missing that.

In fact I don't understand the lines 27-29:
```
  let poly, env' =
    match cb.const_universes with
    | Monomorphic ctx -> false, env
```
Why is the env not enriched in the monomorphic case? Are the universes declared in the definition supposed to be yet here?
If so, why do we take `env'` instead of `env` in line 61?
```
  let env =
    if poly then add_constant kn cb env
    else add_constant kn cb env'
```

Reply:
>This should be in the let env', such that you don't need to reset typing flags.

Actually never mind, we already do this resetting manipulation before the PR for the oracle so I'll clean it up in a follow up.